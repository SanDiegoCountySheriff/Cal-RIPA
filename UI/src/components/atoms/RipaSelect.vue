<template>
  <v-select
    v-model="model"
    clerable
    dense
    flat
    :item-text="itemText"
    :item-value="itemValue"
    :label="label"
    required
    :items="items"
    :rules="rules"
  ></v-select>
</template>

<script>
export default {
  name: 'ripa-select',

  data() {
    return {
      viewModel: this.value,
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
      set(newVal) {
        this.viewModel = newVal
        this.$emit('input', newVal)
      },
    },
  },

  watch: {
    value(newVal) {
      this.viewModel = newVal
    },
  },

  props: {
    value: {
      type: String || Number,
      default: null,
    },
    items: {
      type: Array,
      default: () => [],
    },
    itemText: {
      type: String,
      default: '',
    },
    itemValue: {
      type: String,
      default: '',
    },
    hint: {
      type: String,
      default: '',
    },
    label: {
      type: String,
      default: '',
    },
    rules: {
      type: Array,
      default: () => [],
    },
  },
}
</script>
