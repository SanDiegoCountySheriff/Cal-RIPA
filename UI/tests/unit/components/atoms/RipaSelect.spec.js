import RipaSelect from '@/components/atoms/RipaSelect'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Select', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {})
  })

  const factory = propsData => {
    return shallowMount(RipaSelect, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaSelect, { vuetify })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should set model', async () => {
    wrapper = factory()

    wrapper.vm.model = 'New Value'
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.viewModel).toEqual('New Value')
    expect(wrapper.emitted('input')).toBeTruthy()
  })

  it('should watch value', async () => {
    wrapper = factory()

    wrapper.vm.value = 'New Value'
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.viewModel).toEqual('New Value')
  })
})
