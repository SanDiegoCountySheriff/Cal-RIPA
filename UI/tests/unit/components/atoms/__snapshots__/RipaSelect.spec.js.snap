// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Select should match snapshot 1`] = `<v-select-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" validateonblur="true" appendicon="$dropdown" backgroundcolor="" label="" loaderheight="2" clearicon="$clear" flat="true" type="text" valuecomparator="[Function]" nodatatext="$vuetify.noDataText" items="" itemcolor="primary" itemdisabled="disabled" itemtext="" itemvalue="" menuprops="[object Object]" clerable="" required=""></v-select-stub>`;
